models/Document/Service.php

```diff
@@ -94,39 +94,6 @@ public static function render(Document\PageSnippet $document, array $attributes
         return $content;
     }
 
-    /**
-     * Save document and all child documents
-     *
-     * @param Document $document
-     * @param int $collectGarbageAfterIteration
-     * @param int $saved
-     *
-     * @throws \Exception
-     */
-    private static function saveRecursive($document, $collectGarbageAfterIteration = 25, &$saved = 0)
```

Comment:
Should revert the visibility to `public` and keep it? we also have copyRecursive in the same service which is not used internally.

Reply:
This method was marked pirvate in April 2021:
https://github.com/pimcore/pimcore/pull/8899
![image](https://user-images.githubusercontent.com/38959016/226269994-5aee090b-5f72-4a48-81e0-520b3cd56f1b.png)
I don't think we should keep it.

I also saw that copyRecursive is only being used by itself, but I was not sure since it is public and therefore would not be considered a problem for stan. Would be then out of the scope of this PR right?

Reply:
`copyRecursive` is consistent for DataObject\Asset\Document services so yes, it is a different topic.